### PR TITLE
Visible Bell/Audible Bell options for terminal

### PR DESCRIPTION
**Pull request by christopherdumas**
Also, the windows can now be opened more than once (and save their state in between).

### CODE REVIEW

File: Applications/Terminal/main.cpp [hunk 1]
```diff
@@ -106,6 +106,7 @@ int main(int argc, char** argv)
     terminal.apply_size_increments_to_window(*window);
     window->show();
     window->set_icon_path("/res/icons/16x16/app-terminal.png");
+    terminal.set_should_beep(config->read_num_entry("Window", "AudibleBeep", 1) == 1);
```
Comment (awesomekling) [inline Applications/Terminal/main.cpp]: Let's use `read_bool_entry()` instead.

File: Applications/Terminal/compile_commands.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+[
```
Comment (awesomekling) [inline Applications/Terminal/compile_commands.json]: This looks like an accidental inclusion.

File: Applications/Terminal/Terminal.cpp [hunk 3]
```diff
@@ -586,7 +586,10 @@ void Terminal::on_char(byte ch)
         }
         return;
     case '\a':
-        sysbeep();
+        if (m_should_beep)
+            sysbeep();
+        else
+            m_visual_beep_frames = 2;
```
Comment (awesomekling) [inline Applications/Terminal/Terminal.cpp]: What's driving the next 2 repaints after this? Just the cursor blinking or random I/O occurring?
Seems like we could get stuck in "visual beep mode" for quite a long time.

I would suggest something like this instead:
1. Set some kind of `m_beeping_visually` flag.
2. Call `update()` so we repaint soon.
3. Spawn a single-shot CTimer with some short timeout. In the callback, clear the `m_beeping_visually` flag and call `update()` once again.

File: Applications/Terminal/main.cpp [hunk 4]
```diff
@@ -124,15 +125,35 @@ int main(int argc, char** argv)
...
+    auto* radio_buttons = new GWidget;
+    beep_choice_window->set_main_widget(radio_buttons);
+    radio_buttons->set_fill_with_background_color(true);
+    radio_buttons->set_layout(make<GBoxLayout>(Orientation::Vertical));
+    radio_buttons->layout()->set_margins({ 4, 4, 4, 4 });
+
+    auto* sysbell_radio = new GRadioButton("Use (Audible) System Bell", radio_buttons);
+    auto* visbell_radio = new GRadioButton("Use (Visual) Terminal Bell", radio_buttons);
+    sysbell_radio->set_checked(terminal.should_beep());
+    sysbell_radio->on_checked = [&terminal] (const bool res) {
```
Comment (awesomekling) [inline Applications/Terminal/main.cpp]: What does `res` mean? I would call this `bool checked`.

File: Applications/Terminal/main.cpp [hunk 5]
```diff
@@ -124,15 +125,35 @@ int main(int argc, char** argv)
     slider->set_range(0, 100);
     slider->set_value(100);
 
+    auto* beep_choice_window = new GWindow;
+    beep_choice_window->set_title("Terminal beep settings");
```
Comment (awesomekling) [inline Applications/Terminal/main.cpp]: We should eventually merge this with the opacity settings and whatever else we can think of into a single settings window.
It's not crucial that we do it now though :)

File: Applications/Terminal/main.cpp [hunk 6]
```diff
@@ -124,15 +125,35 @@ int main(int argc, char** argv)
...
 
     auto menubar = make<GMenuBar>();
 
     auto app_menu = make<GMenu>("Terminal");
     app_menu->add_action(GAction::create("Adjust opacity...", [opacity_adjustment_window] (const GAction&) {
-        opacity_adjustment_window->show();
-    }));
+                                                                  opacity_adjustment_window->show();
+                                                              }));
+    app_menu->add_action(GAction::create("Change audio output...", [beep_choice_window] (const GAction&) {
```
Comment (awesomekling) [inline Applications/Terminal/main.cpp]: `audio output` is a pretty strange name for this, how about `beep settings` so we are consistent with the window title? :)

**Comment by christopherdumas** [comment]
Okay, that's done.

**Comment by awesomekling** [comment]
Stellar! That audible bell was driving me crazy too, tbh :)

**Pull request merged**